### DISCUSSION TITLE
Sorry for the stupid question, but can mint-lang be used to create github pages?

### DISCUSSION THREAD

**Discussion by kojix2** [mint-lang/mint#543] [Answered] [answered]
Hi!
I am not at all familiar with creating web applications. So, this may be a silly question, but is it possible to create a github page using mint-lang? I could not find such an example, so it is probably not possible. But is there any way to do this?

**Comment by gdotdesign** [comment, marked as answer]
Hi @kojix2,

It is possible to create a GitHub page with Mint, roughly these are the steps that you need to take:

- Create your application in Mint. Routes should use hashtags, since GitHub pages don't support redirects that are needed for other routes.
- Build the app using `mint build --skip-service-worker --relative` which will create all files you need to host in the `dist` directory
- Push the generated files to `gh-pages` branch and active it in the repository settings (https://blog.ediri.io/how-to-create-a-github-gh-pages-branch-in-an-existing-repository)
- Done :)

Let me know if you have questions!

**Reply by kojix2** [reply]
Hi @gdotdesign 

Thanks for your quick reply.  I actually tried it.
1. created a [project](https://github.com/kojix2/mint_practice) with mint init command.
2. compiled the project using [github actions](https://github.com/kojix2/mint_practice/blob/main/.github/workflows/pages.yml).
3. configured the [gh-pages branch](https://github.com/kojix2/mint_practice/tree/gh-pages) for Github Pages.

However, nothing appears in my browser.  What am I missing?

https://kojix2.github.io/mint_practice/

**Reply by gdotdesign** [reply]
Sorry I miswrote the build command, you need to add the `--relative` flag, since now it tried to load the `index.js` from the root domain, so the full build command is:

```
mint build --skip-service-worker --relative
```

**Reply by kojix2** [reply]
Great! It worked!
I appreciate it.
